crates/matrix-sdk-appservice/src/lib.rs

```diff
@@ -155,29 +164,31 @@ impl AppService {
     ///   with the homeserver.
     ///
     /// [AppService Registration]: https://matrix.org/docs/spec/application_service/r0.1.2#registration
-    pub async fn new(
-        homeserver_url: impl TryInto<Url, Error = url::ParseError>,
-        server_name: impl TryInto<OwnedServerName, Error = IdParseError>,
+    pub fn new(
+        homeserver_url: Url,
+        server_name: OwnedServerName,
         registration: AppServiceRegistration,
```

Comment:
Shouldn't these be setter functions in the builder-pattern though? Like one important  benefit is that you don't have to edit fn signatures for adding new params in the build-pattern, but here, you do have to..

Reply:
I tried to follow crate conventions by having required parameters in the builder's constructor (See f.x. the `VirtualUserBuilder` in the appservice crate), but I'm happy to change it to a more conventional builder pattern. What would you prefer?

Reply:
I think it's probably good to merge as-is, and we can revisit later. This is indeed what we have in most places, also when constructing a `LoginBuilder` through one of the login methods on `Client`.

Sounds good to you, @gnunicorn?

Reply:
yeah, later is fine, hence me giving a thumbs up.